src/Management/Resources/SearchEngines.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Doofinder\Management\Resources;
+
+use Doofinder\Configuration;
+use Doofinder\Shared\Interfaces\HttpClientInterface;
+
+class SearchEngines
+{
+    private $httpClient;
+    private $config;
+
+    private function __construct(HttpClientInterface $httpClient, Configuration $config)
+    {
+        $this->httpClient = $httpClient;
```

Comment:
If this resource accepts the client instance, the others (`Items` and `Indices`) shall also accept them isn't it?

Depends of how we will implement the solution, I mean, the client's could be the one that have the function that serialize and send the resources via http calls, or the resources could have the client instance and do the serialization and data sending

@agutierrezrodriguez what do you suggest?

Reply:
You are right! I forgot the other parameters in Items, Indices,...

I created the resources classes because to delegate the responsibility of data serialization and data sending, but I'm open to do that in a different way.

Reply:
I like the chance of delegate the responsibility of data serialization and manipulation to another classes. Client only should manage httpclient, options and derive the functions calls to the propper classes (for serialization and error management).

Reply:
You can have a generic constructor in a "base resource" class, and custom constructors per resource if it is required. The base resource can have the config param and any base params... But it could be for next pr's also if @agutierrezrodriguez agrees

Reply:
In this case I think it's a good idea. :thinking: